.builder/actions/aws_crt_python.py

```diff
@@ -27,6 +30,45 @@ def run(self, env):
         return Builder.Script(steps, name='install-python-reqs')
 
 
+class SetupForTests(Builder.Action):
+
+    def run(self, env):
+        # we want to use boto3 to pull data from secretsmanager
+        # boto3 might need to be installed first...
+        try:
+            import boto3
+        except BaseException:
+            subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'boto3'])
+            import boto3
+
+        secrets = boto3.client('secretsmanager')
+
+        # get string from secretsmanager and store in environment variable
+        def setenv_from_secret(env_var_name, secret_name):
+            response = secrets.get_secret_value(SecretId=secret_name)
+            env.shell.setenv(env_var_name, response['SecretString'])
+
+        # get file contents from secretsmanager, store as file under /tmp
+        # and store path in environment variable
+        def setenv_tmpfile_from_secret(env_var_name, secret_name, file_name):
+            response = secrets.get_secret_value(SecretId=secret_name)
+            file_contents = response['SecretString']
+            file_path = os.path.join(tempfile.gettempdir(), file_name)
+            pathlib.Path(file_path).write_text(file_contents)
+            env.shell.setenv(env_var_name, file_path)
+
+        setenv_from_secret('AWS_TEST_IOT_MQTT_ENDPOINT', 'unit-test/endpoint')
+
+        setenv_tmpfile_from_secret('AWS_TEST_TLS_CERT_PATH', 'unit-test/certificate', 'certificate.pem')
+        setenv_tmpfile_from_secret('AWS_TEST_TLS_KEY_PATH', 'unit-test/privatekey', 'privatekey.pem')
+
+        # tests must run with leak detection turned on
+        env.shell.setenv('AWS_CRT_MEMORY_TRACING', '2')
+
+        # enable S3 tests
+        env.shell.setenv('AWS_TEST_S3', '1')
```

Comment:
Just thoughts: Is it worth to put those environment variables to README?
- it will be easier for people to know about. 
- But, we do have warnings when we skip tests about the env variables. Hmmm.